text/0000-javascript-module-api.md

```diff
@@ -0,0 +1,1058 @@
+| `Ember.deprecateFunc`                 | `import { deprecateFunc } from "@ember/application/deprecations"`          |
+| `Ember.get`                           | `import { get } from "@ember/object"`                                      |
+| `Ember.getOwner`                      | `import { getOwner } from "@ember/object/internals"`                       |
+| `Ember.getProperties`                 | `import { getProperties } from "@ember/object"`                            |
+| `Ember.guidFor`                       | `import { guidFor } from "@ember/object/internals"`                        |
+| `Ember.inject.controller`             | `import { inject } from "@ember/controller"`                               |
+| `Ember.inject.service`                | `import { inject } from "@ember/service"`                                  |
+| `Ember.inspect`                       | `import { inspect } from "@ember/debug"`                                   |
+| `Ember.instrument`                    | `import { instrument } from "@ember/instrumentation"`                      |
+| `Ember.isArray`                       | `import { isArray } from "@ember/array"`                                   |
+| `Ember.isBlank`                       | `import { isBlank } from "@ember/utils"`                                   |
+| `Ember.isEmpty`                       | `import { isEmpty } from "@ember/utils"`                                   |
+| `Ember.isNone`                        | `import { isNone } from "@ember/utils"`                                    |
+| `Ember.isPresent`                     | `import { isPresent } from "@ember/utils"`                                 |
+| `Ember.keys`                          | `import { keys } from "@ember/polyfills"`                                  |
+| `Ember.makeArray`                     | `import { makeArray } from "@ember/array"`                                 |
+| `Ember.observer`                      | `import { observer } from "@ember/object"`                                 |
+| `Ember.on`                            | `import { on } from "@ember/object/evented"`                               |
+| `Ember.onLoad`                        | `import { onLoad } from "@ember/application"`                              |
+| `Ember.platform.defineProperty`       | `import { defineProperty } from "@ember/polyfills"`                        |
+| `Ember.platform.hasPropertyAccessors` | `import { hasPropertyAccessors } from "@ember/polyfills"`                  |
+| `Ember.removeListener`                | `import { removeListener } from "@ember/object/events"`                    |
+| `Ember.removeObserver`                | `import { removeObserver } from "@ember/object/observers"`                 |
+| `Ember.reset`                         | `import { reset } from "@ember/instrumentation"`                           |
+| `Ember.run`                           | `import { run } from "@ember/runloop"`                                     |
```

Comment:
I think the answer to that would have to be "no" for this and other packages, because constructing the object would require us to eagerly import everything that was tacked on as a property. However, the codemod (should) already handle this case and import the most-deeply-nested part of the property path correctly (e.g. `Ember.run.scheduleOnce` gets replaced with `import { scheduleOnce } from "@ember/runloop"` instead of `import { run } from "@ember-runloop"`).